cmd/issue_show.go

```diff
@@ -118,6 +122,129 @@ WebURL: %s
 	)
 }
 
+// maxPadding returns the max value of two string numbers
+func maxPadding(xstr string, ystr string) int {
+	x, _ := strconv.Atoi(xstr)
+	y, _ := strconv.Atoi(ystr)
+	if x > y {
+		return len(xstr)
+	}
+	return len(ystr)
+}
+
+// printDiffLine does a color print of a diff lines.  Red lines are removals
+// and green lines are additions.
+func printDiffLine(strColor string, maxChars int, sOld string, sNew string, ltext string) {
+
+	switch strColor {
+	case "":
+		fmt.Printf("%*s %*s %s\n", maxChars, sOld, maxChars, sNew, ltext)
+	case "green":
+		color.Green("%*s %*s %s\n", maxChars, sOld, maxChars, sNew, ltext)
+	case "red":
+		color.Red("%*s %*s %s\n", maxChars, sOld, maxChars, sNew, ltext)
+	}
+}
+
+// displayDiff displays the diff referenced in a discussion
+func displayDiff(diff string, chunkNum int, newLine int, oldLine int) {
+	var (
+		diffChunkNum int = -1
+		oldLineNum   int = 0
+		newLineNum   int = 0
+		maxChars     int
+	)
+
+	scanner := bufio.NewScanner(strings.NewReader(diff))
+	for scanner.Scan() {
+		if regexp.MustCompile(`^@@`).MatchString(scanner.Text()) {
+			diffChunkNum++
+			s := strings.Split(scanner.Text(), " ")
+			dOld := strings.Split(s[1], ",")
+			dNew := strings.Split(s[2], ",")
+			maxChars = maxPadding(dOld[1], dNew[1]) + 1
```

Comment:
To keep my mission of breaking your code [ :))) ]: while testing in a private repo, I commented to a patch, which generated the comment with the diffs on the MR "overview" tab, I did run `lab mr show --comments 2` and then:
```
-----------------------------------
#421399302: bmeneg started a commit discussion at 2020-09-30 14:17:13.466 +0000 UTC

File:file
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/zaquestion/lab/cmd.displayDiff(0xc0005dca10, 0x66, 0x0, 0x1, 0x0)
        /home/bmeneg/git/lab/cmd/issue_show.go:165 +0x7cb
github.com/zaquestion/lab/cmd.displayCommitDiscussion(0xc000350a80)
        /home/bmeneg/git/lab/cmd/issue_show.go:243 +0x20a
github.com/zaquestion/lab/cmd.PrintDiscussions(0xc000ba0780, 0x13, 0x13, 0x0, 0x0, 0xfcc876, 0x2, 0x2)
        /home/bmeneg/git/lab/cmd/issue_show.go:311 +0x8e9
github.com/zaquestion/lab/cmd.glob..func29(0x18d0b00, 0xc00064e2c0, 0x1, 0x2)
        /home/bmeneg/git/lab/cmd/mr_show.go:80 +0x536
github.com/spf13/cobra.(*Command).execute(0x18d0b00, 0xc00064e2a0, 0x2, 0x2, 0x18d0b00, 0xc00064e2a0)
        /home/bmeneg/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:846 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0x18cb460, 0xc0000320a0, 0x5, 0x5)
        /home/bmeneg/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
        /home/bmeneg/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/zaquestion/lab/cmd.Execute()
        /home/bmeneg/git/lab/cmd/root.go:345 +0x224
main.main()
        /home/bmeneg/git/lab/main.go:28 +0x93
```

I didn't dig into the issue :/, but it was introduced by this patchset, since on `master` it doesn't happen.

Reply:
Good find @bmeneguele !

The patch can have, for example, either

`@@ -1 +1 @@`

or
`
@@ -1272,6 +1272,8 @@`

I didn't realize the first example (which is what you have in your test tree) could exist.  I've fixed the code and included a little note about the difference in patch output directly in the code.

Reply:
And I say the same: good finding on the @@ line :+1:
I tested here the same case and it's now working.